core/kazoo_data/src/kzs_cache.erl

```diff
@@ -262,3 +267,157 @@ flush_cache_docs(Db, Docs, Options) ->
+        {PInfoKey, N} when N < MaxMsgs ->
+            'ok';
+        {PInfoKey, N} ->
+            lager:error("Cache already has ~p queued messages, exiting (~p loops)",
+                        [N, LoopCount]),
+            Parent ! {'loops', self(), LoopCount},
+            exit('normal')
+    end,
+    %% Insert a key/value pair into the cache.
+    DocId = kz_binary:rand_hex(16),
+    Type = 'undefined',
+    Category = <<"configuration">>,
+    AMQPPayload = load_test_build_amqp_payload(<<"doc_created">>, DbName, Type, DocId, Category),
+    Payload = #cache_obj{'key'=DocId
+                        ,'value'=AMQPPayload
+                        ,'expires'=1000
+                        ,'callback'='undefined'
+                        ,'origin'={'db', DbName, DocId}
+                        },
+    CachePid ! {'$gen_call', {self(), make_ref()}, Payload},
+    %% Flush the agent's mailbox (because of the '$gen_call').
+    'ok' = receive after 0 -> 'ok' end,
+    %% "doc edited"
+    'ok' = load_test_publish(CachePid, <<"doc_edited">>, DbName, Type, DocId, Category),
+    load_test_agent(CachePid, DbName, MaxMsgs, Parent, LoopCount + 1).
+
+-spec wait_for([] | [{pid(), reference()}]) -> 'ok' | 'no_return'.
+wait_for([]) ->
+    'ok';
+wait_for([{Pid, Ref} | Spawns]) ->
+    receive
+        {'DOWN', Ref, 'process', Pid, 'normal'} ->
+            wait_for(Spawns)
+        after 1000 ->
+            lager:error("~p failed to return, ~p workers left.", [Pid, length(Spawns)]),
+            exit("Failed to return")
+    end.
+
+-spec collect_loops(non_neg_integer()) -> pos_integer().
+collect_loops(CurrentCount) ->
+    receive
+        {'loops', _Pid, LoopCount} ->
+            collect_loops(CurrentCount + LoopCount)
+    after 1000 ->
+        CurrentCount
+    end.
+
+-spec load_test_publish(pid()
+                       ,kz_term:ne_binary()
+                       ,kz_term:ne_binary()
+                       ,kz_term:ne_binary()
+                       ,kz_term:ne_binary()
+                       ,kz_term:ne_binary()) -> 'ok'.
+load_test_publish(CachePid, EvName, Db, EvType, Id, Category) ->
+    RKey = binary:list_to_bin(lists:join(<<".">>, [EvName, Db, kz_term:to_binary(EvType), Id])),
+    BD = load_test_build_bd(Category, RKey),
+    PBasic = load_test_build_pbasic(),
+    Payload = load_test_build_amqp_payload(EvName, Db, EvType, Id, Category),
+    CachePid ! {BD, #amqp_msg{'props'=PBasic, 'payload'=kz_json:encode(Payload)}},
+    'ok'.
+
+-spec load_test_build_bd(kz_term:ne_binary(), kz_term:ne_binary()) -> #'basic.deliver'{}.
+load_test_build_bd(Category, RKey) ->
+    #'basic.deliver'
+    {'consumer_tag' = kz_binary:rand_hex(16)
+    ,'delivery_tag' = rand:uniform(1000)
+    ,'exchange' = Category
+    ,'routing_key' = RKey
+    }.
+
+-spec load_test_build_pbasic() -> #'P_basic'{}.
+load_test_build_pbasic() ->
+    #'P_basic'
+    {'content_type' = <<"application/json">>
+    ,'content_encoding' = 'undefined'
```

Comment:
Since these are records, the `undefined` value is implied; remove these until such time as they are needed.

Reply:
Done, thanks 👍